src/osal/files_win32.c

```diff
@@ -152,43 +156,46 @@ const char * osal_get_shared_filepath(const char *filename, const char *firstsea
 
 const char * osal_get_user_configpath(void)
 {
-    static char chHomePath[MAX_PATH];
+    static wchar_t chHomePath[MAX_PATH];
+    static char outString[MAX_PATH];
     LPITEMIDLIST pidl;
     LPMALLOC pMalloc;
     struct _stat fileinfo;
 
     // Get item ID list for the path of user's personal directory
     SHGetSpecialFolderLocation(NULL, CSIDL_APPDATA, &pidl);
     // get the path in a char string
-    SHGetPathFromIDList(pidl, chHomePath);
+    SHGetPathFromIDListW(pidl, chHomePath);
     // do a bunch of crap just to free some memory
     SHGetMalloc(&pMalloc);
     pMalloc->lpVtbl->Free(pMalloc, pidl);
     pMalloc->lpVtbl->Release(pMalloc);
 
     // tack on 'mupen64plus'
-    if (chHomePath[strlen(chHomePath)-1] != '\\')
-        strcat(chHomePath, "\\");
-    strcat(chHomePath, "Mupen64Plus");
+    if (chHomePath[wcslen(chHomePath)-1] != L'\\')
+        wcscat(chHomePath, L"\\");
+    wcscat(chHomePath, L"Mupen64Plus");
 
     // if this directory doesn't exist, then make it
-    if (_stat(chHomePath, &fileinfo) == 0)
+    if (_wstat(chHomePath, &fileinfo) == 0)
     {
-        strcat(chHomePath, "\\");
-        return chHomePath;
+        wcscat(chHomePath, L"\\");
+        WideCharToMultiByte(CP_UTF8, 0, chHomePath, -1, outString, MAX_PATH, NULL, NULL);
+        return outString;
     }
     else
     {
-        osal_mkdirp(chHomePath, 0);
-        if (_stat(chHomePath, &fileinfo) == 0)
+        _wmkdir(chHomePath);
```

Comment:
the only problem I have with this PR is this.  the osal_mkdirp() function will create any necessary new folders in the full path, but _wmkdir() will only create one new folder (the last item in the given path).  One solution for this would be to add a new static function in files_win32.c called _wmkdirp() or something like that.

Reply:
This was probably just an oversight on my part, I don't think there is any reason that this can't be osal_mkdirp, since I updated the internals of that function to use _wmkdir for windows. I'll test it tomorrow

Reply:
okay, good. just be careful because osal_mkdirp() might expect utf8 input but I think chHomePath has already been converted to wchar here.

Reply:
Ok I took care of this, now it's calling ```osal_mkdirp``` (after converting the string to utf-8)